File: java8/src/test/java/PathsTest.java
import org.junit.Assert;
import org.junit.Test;

public class PathsTest {
    final Point a = new Point(0, 10);
    final Point b = new Point(0, 20);
    final Segment ab = new Segment(a, b);

    @Test
    public void testSegmentDistance0() {
        Segment segment = new Segment(a, a);
        double distance = 0; // TODO
        Assert.assertEquals(0, distance, 0.01);
    }

}

File: kotlin/quickstart/src/test/kotlin/PathsTest.kt
import org.junit.Assert
import org.junit.Test

class PathsTest {
    val a = Point(0, 10)
    val b = Point(0, 20)
    val c = Point(10, 20)
    val ab = Segment(a, b)
    val ac = Segment(a, c)
    val bc = Segment(b, c)

    @Test
    fun testSegmentDistance0() {
        val segment = Segment(a, a)
        Assert.assertEquals(0.0, segment.distance, 0.01)
    }

    @Test
    fun testSegmentDistanceNonZero() {
        Assert.assertEquals(10.0, ab.distance, 0.01)
    }
}
